android/engine/src/main/java/org/smartregister/fhircore/engine/auditEvent/AuditEventWorker.kt

```diff
@@ -0,0 +1,84 @@
+package org.smartregister.fhircore.engine.auditEvent
+
+import android.content.Context
+import androidx.hilt.work.HiltWorker
+import androidx.work.CoroutineWorker
+import androidx.work.WorkerParameters
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.logicalId
+import dagger.assisted.Assisted
+import dagger.assisted.AssistedInject
+import dagger.hilt.android.qualifiers.ApplicationContext
+import org.hl7.fhir.r4.model.AuditEvent
+import org.hl7.fhir.r4.model.Coding
+import org.hl7.fhir.r4.model.Practitioner
+import org.smartregister.fhircore.engine.data.local.DefaultRepository
+import org.smartregister.fhircore.engine.util.SharedPreferenceKey
+import org.smartregister.fhircore.engine.util.SharedPreferencesHelper
+import org.smartregister.fhircore.engine.util.extension.asReference
+import timber.log.Timber
+import java.util.Date
+import java.util.UUID
+
+@HiltWorker
+class AuditEventWorker
+@AssistedInject
+constructor(
+    @Assisted val appContext: Context,
+    @Assisted workerParameters: WorkerParameters,
+    val fhirEngine: FhirEngine,
+    val sharedPreferences: SharedPreferencesHelper,
+    val defaultRepository: DefaultRepository
+    ) : CoroutineWorker(appContext, workerParameters) {
+
+    override suspend fun doWork(): Result {
+        Timber.e("AuditEventWorker is running")
+        createAuditEvent()
+        return Result.success()
+    }
+
+    private suspend fun createAuditEvent(){
+        // Get Practitioner Resource
+        val practitionerID =
+            sharedPreferences.read(key = SharedPreferenceKey.PRACTITIONER_ID.name, defaultValue = null)
+        val practitioner = defaultRepository.loadResource<Practitioner>(practitionerID!!)
+
+        // Create AuditEvent Resource
+        val auditEvent = AuditEvent().apply {
```

Comment:
Let's have an AuditRepository that has the code for creating the audit events. You can inject that and use it